### PR TITLE
Made IRowMapper take IRow for metadata

### LINKED ISSUE #719: IRowMapper to use IRow for metadata
Change the interface of `IRowMapper` to use `IColumn` (or something akin to it) to convey the schema information. 

This will simplify the usage of `IRowMapper` in cases where it emits metadata for the new columns.

### PR DESCRIPTION
**Pull request by Zruty0**
Fixes #719

### CODE REVIEW

**Comment by Zruty0** [comment]
@TomFinley  @yaeldekel  please kindly take a look

**Comment by Ivanidzo4ka** [comment]
Can you bump your PR, build system is stuck for some reason.

File: src/Microsoft.ML.Data/DataView/RowToRowMapperTransform.cs [hunk 1]
```diff
@@ -15,13 +15,31 @@
...
+
     public sealed class RowMapperColumnInfo
     {
         public readonly string Name;
         public readonly ColumnType ColType;
-        public readonly ColumnMetadataInfo Metadata;
+        public readonly IRow Metadata;
 
-        public RowMapperColumnInfo(string name, ColumnType type, ColumnMetadataInfo metadata)
+        public RowMapperColumnInfo(string name, ColumnType type, IRow metadata)
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Data/DataView/RowToRowMapperTransform.cs]: >IRow [](start = 65, length = 4)

Any chance you willing to modify one of transforms to show how it's gonna be used?

**Comment by TomFinley** [comment]
>     public interface IColumn

The parallel interfaces are a little unfortunate. What do you think about making `IColumn` descend from your new `IColumnInfo` interface (or even putting that interface here)? It's a little strange that `IColumnInfo` interface is hiding in a file `RowToRowMapperTransform.cs` where I certainly would not expect to find it?

IF you did that, you could then remove three of the properties here, which seems like a positive change.

---
Refers to: src/Microsoft.ML.Data/Data/IColumn.cs:32 in 4169339. [](commit_id = 4169339bff1d4b4199d1fed361eccd234bc3d29d, deletion_comment = False)

**Comment by Zruty0** [comment]
>     public interface IColumn

Actually, `IColumnInfo` is not being used, so I'll remove it for now

---
In reply to: [417409062](https://github.com/dotnet/machinelearning/pull/728#issuecomment-417409062) [](ancestors = 417409062)

---
Refers to: src/Microsoft.ML.Data/Data/IColumn.cs:32 in 4169339. [](commit_id = 4169339bff1d4b4199d1fed361eccd234bc3d29d, deletion_comment = False)

**Comment by Zruty0** [comment]
Previous test failure seemed to be a flaky one.

**Pull request merged**